pkg/usersettings/handlers.go

```
@@ -0,0 +1,263 @@
+	context := context.TODO()
+
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	role := &rbac.Role{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "Role",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleName(),
+		},
+		Rules: []rbac.PolicyRule{
+			rbac.PolicyRule{
+				APIGroups: []string{
+					"", // Core group, not "v1"
+				},
+				Resources: []string{
+					"configmaps", // Not "ConfigMap"
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"patch",
+					"update",
+					"watch",
+				},
+				ResourceNames: []string{
+					resourceData.getConfigMapName(),
+				},
+			},
+		},
+	}
+
+	roleBinding := &rbac.RoleBinding{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "RoleBinding",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleBindingName(),
+		},
+		Subjects: []rbac.Subject{
+			rbac.Subject{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "User",
+				Name:     user.Username,
+			},
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     resourceData.getRoleName(),
+		},
+	}
+
+	configMap := &core.ConfigMap{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getConfigMapName(),
+		},
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = client.RbacV1().Roles(USER_SETTINGS_NAMESPACE).Create(context, role, meta.CreateOptions{})
+	if err != nil {
+		klog.Warningf("Could not create user-settings Role \"%s\" for user \"%s\" (UID: %s)", resourceData.getRoleName(), resourceData.Username, resourceData.UID)
+	}
+
+	_, err = client.RbacV1().RoleBindings(USER_SETTINGS_NAMESPACE).Create(context, roleBinding, meta.CreateOptions{})
```

What happens when one of these resources(Role/Rolebinding/SA) fails to get created? Don't think we should proceed with creating the rest of them and leave it in an inconsistent state.

---

Yep, was ignored before. I switched to this code to ignore just an `AlreadyExists` error. Everything else let the call fail and was logged then in `HandleUserSettings`.

```go
	if err != nil && !apierrors.IsAlreadyExists(err) {
		return nil, err
	}
```